wforce/wforce-web.cc

```diff
@@ -211,41 +215,220 @@ void parseSyncCmd(const YaHTTP::Request& req, YaHTTP::Response& resp, const std:
 
   msg = Json::parse(req.body, err);
   if (msg.is_null()) {
-    resp.status=500;
+    resp.status = 500;
     std::stringstream ss;
     ss << "{\"status\":\"failure\", \"reason\":\"" << err << "\"}";
-    resp.body=ss.str();
+    resp.body = ss.str();
   }
   else {
     try {
       if (msg["replication_host"].is_null() ||
           msg["replication_port"].is_null() ||
           msg["callback_url"].is_null() ||
           msg["callback_auth_pw"].is_null()) {
-        throw WforceException("One of mandatory parameters [replication_host, replication_port, callback_url, callback_auth_pw] is missing");
+        throw WforceException(
+            "One of mandatory parameters [replication_host, replication_port, callback_url, callback_auth_pw] is missing");
       }
       string myip = msg["replication_host"].string_value();
       int myport = msg["replication_port"].int_value();
       ComboAddress replication_ca(myip, myport);
       std::string callback_url = msg["callback_url"].string_value();
       std::string callback_pw = msg["callback_auth_pw"].string_value();
-      thread t(syncDBThread, replication_ca, callback_url, callback_pw);
+      std::string encryption_key;
+      if (!msg["encryption_key"].is_null()) {
+        encryption_key = msg["encryption_key"].string_value();
+      }
+      else {
+        encryption_key = g_key;
+      }
+      thread t(syncDBThread, replication_ca, callback_url, callback_pw, encryption_key);
       t.detach();
     }
-    catch(const WforceException& e) {
-      resp.status=500;
+    catch (const WforceException& e) {
+      resp.status = 500;
       std::stringstream ss;
       ss << "{\"status\":\"failure\", \"reason\":\"" << e.reason << "\"}";
-      resp.body=ss.str();
+      resp.body = ss.str();
       errlog("Exception in command [%s] exception: %s", command, e.reason);
     }
   }
   if (resp.status == 200)
-    resp.body=R"({"status":"ok"})";
+    resp.body = R"({"status":"ok"})";
   incCommandStat("syncDBs");
   incPrometheusCommandMetric("syncDBs");
 }
 
+void parseAddSiblingCmd(const YaHTTP::Request& req, YaHTTP::Response& resp, const std::string& command)
+{
+  using namespace json11;
+  Json msg;
+  string err;
+
+  resp.status = 200;
+
+  msg = Json::parse(req.body, err);
+  if (msg.is_null()) {
+    resp.status = 500;
```

Comment:
Maybe a 4xx response code would be more relevant ?

Reply:
Yes agreed.

Reply:
Yes, will fix throughout this file